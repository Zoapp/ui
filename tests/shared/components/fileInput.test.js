import React from "react";
import renderer from "react-test-renderer";

import FileInput from "libs/components/fileInput";

describe("FileInput", () => {
  it("renders correctly", () => {
    const component = renderer.create(<FileInput onChange={jest.fn()} />);

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render children", () => {
    const component = renderer.create(
      <FileInput onChange={jest.fn()}>Upload files</FileInput>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should define accepted files", () => {
    const component = renderer.create(
      <FileInput onChange={jest.fn()} accept="audio/mp3">
        Upload files
      </FileInput>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should allow to select multiple files", () => {
    const component = renderer.create(
      <FileInput onChange={jest.fn()} multiple>
        Upload files
      </FileInput>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
