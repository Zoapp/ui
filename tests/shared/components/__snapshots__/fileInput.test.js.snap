// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInput renders correctly 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <button
    className="mdc-button"
    disabled={false}
    id={null}
    onClick={[Function]}
    type="button"
  >
    
  </button>
  <input
    accept="*/*"
    multiple={undefined}
    onChange={[MockFunction]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;

exports[`FileInput should allow to select multiple files 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <button
    className="mdc-button"
    disabled={false}
    id={null}
    onClick={[Function]}
    type="button"
  >
    
    Upload files
  </button>
  <input
    accept="*/*"
    multiple={true}
    onChange={[MockFunction]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;

exports[`FileInput should define accepted files 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <button
    className="mdc-button"
    disabled={false}
    id={null}
    onClick={[Function]}
    type="button"
  >
    
    Upload files
  </button>
  <input
    accept="audio/mp3"
    multiple={undefined}
    onChange={[MockFunction]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;

exports[`FileInput should render children 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <button
    className="mdc-button"
    disabled={false}
    id={null}
    onClick={[Function]}
    type="button"
  >
    
    Upload files
  </button>
  <input
    accept="*/*"
    multiple={undefined}
    onChange={[MockFunction]}
    style={
      Object {
        "display": "none",
      }
    }
    type="file"
  />
</div>
`;
